### PR TITLE
Handle uploading files to subfolders

### PR DESCRIPTION
**Pull request by hafen**
@hathawayj fixed the files.R and folders.R files to handle uploading files to subfolders and the creation of files in subfolders.

### CODE REVIEW

File: R/files.R [hunk 1]
```diff
@@ -18,18 +18,46 @@
 
 upload_files <- function(id, path, dest = NULL) {
   type <- process_type(id, private = TRUE)
+  subfolder_file = FALSE
```
Comment (hafen) [inline R/files.R]: Change `=` to `<-`

File: R/files.R [hunk 2]
```diff
@@ -18,18 +18,46 @@
...
 
   if (is.null(dest)) {
     dest <- basename(path)
   }
 
-  if (type == 'nodes') {
+  ## Here we need to check for 'nodes' then have another condition within nodes. see `idx_folder`
+ if (type == 'nodes') {
```
Comment (hafen) [inline R/files.R]: Use `"` instead of `'`

File: R/files.R [hunk 3]
```diff
@@ -18,18 +18,46 @@
...
+
     fi <- get_files_info(id, private = TRUE)
     idx <- which(fi$materialized == pre_slash(dest))
-    if (length(idx) != 1) {
+
+    # check for destination value subfolder
+    if(!is.null(dest)) {
```
Comment (hafen) [inline R/files.R]: Use `if (` instead of `if(`

File: R/files.R [hunk 4]
```diff
@@ -18,18 +18,46 @@
...
+
+      hash_folder <- paste0(basename(fi[idx_folder, "href"]), "/")
+
+      # check for subfolder created and file not there.
+      if (length(idx) != 1 & length(idx_folder) == 1) {
+        message(paste0('Creating new file on OSF in subfolder ', fi_folder  ,' ...'))
+        upload_new(id, path, dest_fname, href_hash = hash_folder)
+        subfolder_file = TRUE
+      }
+      if(length(idx_folder) != 1 & dirname(fi_folder) != ".") {
```
Comment (hafen) [inline R/files.R]: Use `if (` instead of `if(`

File: R/files.R [hunk 5]
```diff
@@ -87,12 +115,13 @@ upload_zip <- function(id, path, dest = NULL) {
 #' @param path Path to file on local machine to upload. Ensure file has
 #' proper extension named (i.e., extension sensitive, not like on Linux)
 #' @param name Name of the destination file on OSF (if \code{NULL},
+#' @param href_hash the element that identifies the folder hash of the href from the folder URL. Not use if \code{NULL}.
```
Comment (hafen) [inline R/files.R]: Change "not use" to "not used"

File: R/folders.R [hunk 6]
```diff
@@ -2,11 +2,12 @@
 #'
 #' @param id Parent OSF project id (osf.io/XXXX; just XXXX) to create folder in
 #' @param path Name of the folder (cannot handle recursive at the moment).
-#'
-#' @return Waterbutler URL
+#' @param return Which waterbutler URLs should be returned. Defaults to root to mimick old functions behavior.
```
Comment (hafen) [inline R/folders.R]: Change "mimick" to "mimic"

**Comment by bgrich** [comment]
Hi @hathawayj and @hafen! I am working on merging these changes in currently. Since all of the development is happening on the `mvp` branch, I am going to point these changes to there.

**Comment by bgrich** [comment]
Thanks for your contributions!

**Pull request merged**